rholang/src/main/scala/coop/rchain/rholang/interpreter/error/MLock.scala

```diff
@@ -0,0 +1,41 @@
+package coop.rchain.rholang.interpreter.error
+
+import cats.effect.{Bracket, Concurrent}
+import cats.implicits._
+import cats.effect.concurrent.MVar
+
+final class MLock[F[_]](lock: MVar[F, Unit])(implicit bracketF: Bracket[F, Throwable]) {
```

Comment:
Why implement a lock when you have the less error prone MVar?

Reply:
Their behaviors are different. Take `release` for example. It is an error to release a released lock. I also don't need to expose the entire interface of `MVar`.